src/app/services/canvas.service.ts

```diff
@@ -1289,15 +1462,24 @@ export class CanvasService {
         for (const override of overrides) {
             if (!override.isIndividualLevel) continue;
             if (override.studentIdsAsIndividualLevel.length >= CanvasService.ASSIGNMENT_OVERRIDE_MAX_SIZE) continue;
-            if (
-                this.isOverrideDateEqual(unlockDate, override.unlockAt) &&
-                this.isOverrideDateEqual(lockDate, override.lockAt) &&
-                this.isOverrideDateEqual(lockDate, override.dueAt)
-            ) {
+            if (areOverrideDatesEqual(resultDates, override)) {
```

Comment:
This loop could also be replaced with functional implementation since it is basically `Array.find()` with a list of conditions to check.